stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -171,20 +171,20 @@ size(A::HermOrSym, d) = size(A.data, d)
 size(A::HermOrSym) = size(A.data)
 @inline function getindex(A::Symmetric, i::Integer, j::Integer)
     @boundscheck checkbounds(A, i, j)
-    @inbounds if (A.uplo == 'U') == (i < j)
-        return A.data[i, j]
-    elseif i == j
+    @inbounds if i == j
         return symmetric(A.data[i, j], Symbol(A.uplo))::symmetric_type(eltype(A.data))
```

Comment:
Why the typeassert?  Is that hiding a type-instability?

Reply:
See https://github.com/JuliaLang/julia/pull/25688, so yeah, kinda.

Reply:
No, it's ensuring that `eltype` isn't lying (if `symmetric` does return something else than what `symmetric_type` promised) .